### PR TITLE
Support OPTIONS requests

### PR DESCRIPTION
**Pull request by samfrench**
Support OPTIONS requests

### CODE REVIEW

**Comment by samfrench** [comment]
@dazoakley, can you review this please?

File: lib/alephant/broker/response/asset.rb [hunk 1]
```diff
@@ -17,7 +18,7 @@ def initialize(component, request_env)
         end
 
         def setup
-          @content  = @component.content
+          @content  = (@request_env.options?) ? "" : @component.content
```
Comment (dazoakley) [inline lib/alephant/broker/response/asset.rb]: do you need the `()` here?

File: spec/integration/rack_spec.rb [hunk 2]
```diff
@@ -96,6 +96,33 @@
...
+      specify { expect(last_response.headers["Content-Length"]).to eq("0") }
+    end
+
+    context "for valid URL parameters in request" do
+      before { options "/component/test_component?variant=test_variant" }
+      specify { expect(last_response.status).to eq 200 }
+      specify { expect(last_response.body).to eq "" }
+      specify { expect(last_response.headers["Content-Type"]).to eq("test/content") }
+      specify { expect(last_response.headers["Content-Length"]).to eq("0") }
+    end
```
Comment (dazoakley) [inline spec/integration/rack_spec.rb]: For an invalid endpoint?  Might be worth adding a negative assertion.

**Comment by samfrench** [comment]
@dazoakley, I changed the way the options response was returned, as we need to return an empty string for all options requests, even if they 404 or 500.

**Comment by dazoakley** [comment]
👍

File: lib/alephant/broker/response/base.rb [hunk 3]
```diff
@@ -61,6 +63,10 @@ def add_etag_allow_header
           })
         end
 
+        def self.options?(request_env)
+          request_env && request_env.respond_to?(:options?) && request_env.options?
```
Comment (darnould) [inline lib/alephant/broker/response/base.rb]:
```ruby
request_env &. options?
```
:)

Comment (samfrench) [inline lib/alephant/broker/response/base.rb]: What does `&.` mean? I noticed request_env was sometimes `nil`, so I had to workaround it. After the elections, I'll be able to look at this further, as the logic is all quite tangled, and not straightforward to follow.

Comment (samfrench) [inline lib/alephant/broker/response/base.rb]: Thanks for the suggestion. The more people that take a look, the better.

Comment (dazoakley) [inline lib/alephant/broker/response/base.rb]: Heh, would be nice to use that feature (http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/) but we're currently targeting older builds of ruby. :)

**Pull request merged**